gymnasium/spaces/utils.py

```diff
@@ -102,7 +107,9 @@ def _flatdim_text(space: Text) -> int:
 
 
 T = TypeVar("T")
-FlatType = Union[np.ndarray, TypingDict, tuple, GraphInstance]
+FlatType = Union[
```

Comment:
`NDArray | dict[str, Any] | tuple | GraphInstance`

I'm not 100% sure if this is the actual signature we want for `flatten`, but it will be yet another thing to look at later

Reply:
Agreed